Autocomplete results - ID retrieving fixed

valkars opened the pull request:
<!-- THE PR TEMPLATE IS NOT AN OPTION. DO NOT DELETE IT, MAKE SURE YOU READ AND EDIT IT! -->
## Subject

<!-- Describe your Pull Request content here -->
In the autocomplete response, each item is returned as an array in the format:
```
['id' => $admin->id($model), 'label' => $label]
```
However, the `$admin` variable refers to the current Admin class where the autocomplete filter is used (e.g. `Products`), not the Admin class of the related entity being searched (e.g. `Categories`).

The correct ID should be generated using the `$targetAdmin` variable, which represents the actual Admin class for the autocomplete target.

**Fix**:
Replaced `$admin->id($model)` with `$targetAdmin->id($model)` to ensure correct ID resolution in autocomplete results.

I am targeting this branch, because everything backwards compatible.


## Changelog

<!-- MANDATORY
    Fill the changelog part inside the code block.
    Follow this schema: https://keepachangelog.com/
    This will end up on https://github.com/sonata-project/SonataAdminBundle/releases,
    please keep it short and clear and to the point
-->

```markdown
### Fixed
- ID retrieving in autocomplete results 
```

Commit:
Autocomplete results - ID retrieving fixed

VincentLanglet commented:
Hi @valkars 

Test will need to be updated.

Also, if it's a fix:
- Does it solve an (existing) issue we can link at ?
- Can we add a non regression test ?

Also what is the current impact of the bug ; if the wrong admin is used, it would mean that the id is wrong. Would it mean that autocomplete is not working at all ; this is weird since there is no complaint about it...

valkars commented:
Autocomplete broke 6 years ago in our project. Then I have fixed it by coping this file and using it with my fix. I was lazy to write the issue or made a PR. Now I decided to make a PR. I think that bug was not found, because of obstacles:
Categories have id (int) and uuid (string), we decided to use uuid as id, so I've added:
```
 public function id($model): ?string
{
        return $model?->getUuid()->toString();
}
 public function getIdParameter(): string
{
        return 'uuid';
}
```
So wrong Admin class (Products) returns wrong ID, (id field, int), not uuid value for category and all is broken.
I am lazy to write regression tests, I can use my fix further :) But I'll take a look on failing tests

VincentLanglet commented:
Thanks
